src/ModelContextProtocol/Shared/McpSession.cs

```diff
@@ -63,7 +63,7 @@ public McpSession(
         ITransport transport,
         string endpointName,
         RequestHandlers requestHandlers,
-        NotificationHandlers notificationHandlers,
+        Dictionary<string, List<Func<JsonRpcNotification, Task>>> notificationHandlers,
```

Comment:
If we're going to delete NotificationHandlers, we might as well do the same with RequestHandlers. Why not use the sealed derived type though? One of my big motivations for making RequestHandlers and NotificationHandlers was to make this constructor signature shorter.

Like with the capabilities cloning, I'm more concerned with consistency versus what choice we make, but I hope you stick with my way! 😆

Reply:
> Why not use the sealed derived type though?

The type would be empty. You'd just be looking to derive from it just to give it a different name?
```C#
internal sealed class NotificationHandlers : Dictionary<string, List<Func<JsonRpcNotification, Task>>>;
```
?

Reply:
I'd be fine with that. You could also move the contents of McpEndpoint.AddNotificationHandler and AddNotificationHandlers back into NotificationHandlers if it'd make you feel better about it. It also cleanly shows that these methods operate on nothing other than the dictionary and its arguments.

Reply:
> RequestHandlers at least has additional functionality it's exposing.

TBF it's trivial relatively speaking and there's a bit of duplication in forwarding the handler subscription method. I'd be in favour of flattening both.